hoot-services/src/main/java/hoot/services/validators/osm/ChangesetErrorChecker.java

```diff
@@ -57,393 +57,309 @@
+    }
+
+    /**
+     * @throws Exception
+     */
+    public void checkForVersionErrors() throws Exception {
+        log.debug("Checking for element version errors...");
+
+        for (ElementType elementType : ElementType.values()) {
+            if (elementType != ElementType.Changeset) {
+                for (EntityChangeType entityChangeType : EntityChangeType.values()) {
+                    NodeList elementXmlNodes =
+                            XPathAPI.selectNodeList(changesetDoc,
+                                    "//osmChange/" + entityChangeType.toString().toLowerCase() + "/" +
+                                            elementType.toString().toLowerCase());
+
+                    Map<Long, Long> elementIdsToVersionsFromChangeset = new HashMap<>();
+                    for (int i = 0; i < elementXmlNodes.getLength(); i++) {
+                        long id = Long.parseLong(elementXmlNodes.item(i).getAttributes().getNamedItem("id").getNodeValue());
+                        long parsedVersion = Long.parseLong(elementXmlNodes.item(i).getAttributes().getNamedItem("version").getNodeValue());
+                        if (entityChangeType == EntityChangeType.CREATE) {
+                            if (parsedVersion != 0) {
+                                throw new Exception("Invalid version: " + parsedVersion + " specified for created " +
+                                                    elementType + " with ID: " + id + "; expected version 0.");
+                            }
+                        }
+                        else {
+                            elementIdsToVersionsFromChangeset.put(id, parsedVersion);
+                        }
+                    }
+
+                    if ((entityChangeType != EntityChangeType.CREATE) && (!elementIdsToVersionsFromChangeset.isEmpty())) {
+                        Element prototype = ElementFactory.getInstance().create(mapId, elementType, dbConn);
+                        Map<Long, Long> elementIdsToVersionsFromDb =
+                                new SQLQuery(dbConn, DbUtils.getConfiguration(mapId))
+                                        .from(prototype.getElementTable())
+                                        .where(prototype.getElementIdField().in(elementIdsToVersionsFromChangeset.keySet()))
+                                        .map(prototype.getElementIdField(), prototype.getElementVersionField());
+
+                        if (!elementIdsToVersionsFromDb.equals(elementIdsToVersionsFromChangeset)) {
+                            throw new Exception("Invalid version specified for element(s).");
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @throws Exception
+     */
+    public void checkForOwnershipErrors() throws Exception {
+        log.debug("Checking for child element ownership errors...");
+        //checkForRelationOwnershipErrors();
+        //checkForWayOwnershipErrors();
```

Comment:
If these methods are now dead code, they should be removed from the class completely to reduce clutter.